bciers/libs/components/src/form/widgets/BcghgIdWidget.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import { WidgetProps } from "@rjsf/utils/lib/types";
+import { Button } from "@mui/material";
+import { actionHandler } from "@bciers/actions";
+import { useState } from "react";
+import SnackBar from "../components/SnackBar";
+
+export enum EntityWithBcghgType {
+  OPERATION = "operation",
+  FACILITY = "facility",
+}
+
+async function generateBcghgId(
+  entityId: string,
+  entityType: EntityWithBcghgType,
+) {
+  const endpoint =
+    entityType === EntityWithBcghgType.OPERATION
+      ? `registration/v2/operations/${entityId}/bcghg-id`
+      : `registration/facilities/${entityId}/bcghg-id`;
+
+  const response = await actionHandler(endpoint, "PATCH", "");
+
+  return response;
```

Comment:
nit: We can return the non-awaited `actionHandler` because we are doing it when calling the function.